### DISCUSSION TITLE
Salary with multiple expenses

### DISCUSSION THREAD

**Discussion by SingingDwarf** [firefly-iii/firefly-iii#8601] [General]
I know this question has been asked before and the response has been given that Firefly III isn't really made to handle this.

Previous responses have suggested that Firefly is intended to allow a transaction at for example a grocery store to be split, so that the user can see how much has been spent on food, drink, pet food, etc.  I may however also want to record any refunds I received from the grocery store in the same transaction, as it appears on the same receipt as the food, drink and pet food.

The question was asked, why do people want to track a payslip in the same way.

From my perspective, the answer is because it could be much easier to create a single transaction (payslip ≡ supermarket receipt) which includes the net salary credited to my asset account (≡ the total spend at the grocery store), as well as the breakdown (food, drink and pet food). 

In this case, the breakdown would be my base salary (≡ food), bonus (≡ drink), income tax (≡ refund on item returned to supermarket), pension contribution (≡ cashback), etc.

I want to track my base salary, bonuses, pension contributions and income tax individually - ideally within a single transaction, because it is analogous to a supermarket transaction with different items being purchased, refunded, etc.

I guess it is because I am a 20+ year user of MS Money, which has a 'payslip' feature that allows just that. I can split my payslip into wages, deductions before tax, tax and deductions after tax. I enter my total wages and can then add categories under each of the above - so in the UK pension contributions would be taken before tax, whereas gym membership might be taken after tax.

How do you recommend breaking down a payslip with these different components currently - one transaction per entry on the payslip? This is more labour intensive than entering via split transactions - but unfortunately these don't allow a mixture of +ve and -ve splits within the same transaction.

_Originally posted by @JC5 in https://github.com/firefly-iii/firefly-iii/issues/1256#issuecomment-374217892_

**Comment by JC5** [comment]
Thanks for the explanation, this is really insightful.

**Comment by cachho** [comment]
@JC5 @SingingDwarf I was going to open a separate discussion, but I guess the root cause is the same, so let me mention a different scenario. 

Let's say I'm selling something over Paypal.

I want to create a transaction

`
Revenue ($500) -> Paypal Account ($495) / Payment Service Fees ($5)
`

That's the way I learned to do it in double-entry accounting in business school.

The problem is that with a split transaction I can't change the destination account.

edit: explanation: I made $500 revenue for accounting and tax purposes, but not all the money reaches my account because some is lost as an expense.

**Comment by SingingDwarf** [comment]
So no explanation as to how to do this?
Maybe it is an Issue after all?

**Reply by JC5** [reply]
This isn't going to be changed. I appreciate the feedback but this is not on my list. If you want to do this, you'll have to register those parts of your salary as normal transactions.

**Reply by mcmikemn** [reply]
@JC5, does that mean that the recommended way to enter a single paycheck would be 2 transactions? 
1) one _revenue_ transaction with the salary and splits to include things like bonus, stipends, pension contribution, reimbursements from employer, etc.
2) one _expense_ transaction with splits to include federal tax, medicare, social security tax, insurance premiums, etc.

**Comment by SingingDwarf** [comment]
See also #11446
